### ISSUE TITLE
Cook Issues - Stuck on requests and sitting down when hungry

### ISSUE THREAD

**Issue by Jinotad** [ldtteam/minecolonies#7489]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: minecolonies-0.14.328-ALPHA-universal
- Structurize Version: structurize-0.13.208-ALPHA-universal
- Related Mods and their Versions: Adventures 3.5 server

## Expected behavior
When minimum food requests are set, I would expect for the assistant cook to handle all crafting related requests. Allowing for the cook to request all other type of foods for the restaurant. Additionally, I would expect for the cook to work though hunger to ensure food is still being requested / prepared, and eating to at least half food saturation, before serving others.

## Actual behavior
I have put golden carrots and Sweet Bread as minimum 1 stack stock in the restaurant. Occasionally, the cook themselves handle the request for ingredients. If there is not enough ingredients to make the required items, the cook will get stuck waiting on that request to be fulfilled. Resulting in the cook never requesting any other food supplies to keep the restaurant in stock.    

Additionally, if the cook gets hungry and does not have food for themselves, the cook will then sit down and stop all work until they are manually fed.

https://cdn.discordapp.com/attachments/486940969737125922/875071377625976833/Cook_-_Stuck_requesting.mp4

## Steps to reproduce the issue

Setting complicated recipes as minimum stocks in restaurant. Golden Carrots, Loaded Potatoes are the ones I've had primary issues causing this.  Then not have enough ingredients to make the full stack at once, and time.

### Notes
Part of the issue may be that the crafters are attempting to put a request in to make the entire stack at a time, rather than piecemealing what they can.  So currently putting in a single request to craft x64 recipe at once, rather than crafting it in smaller batches such as x8 or x16 at a time.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Jinotad** [comment]
Maybe I've just managed to majorly bug my restaurant?  The only food minimum I have set up at this time, in the restaurant no where else, is plain bread. Yet, after the most recent request came in, this is what my cook is demanding. (After having canceled, and turned off Golden Carrots from the food.) 
![2021-08-12_12 21 53](https://user-images.githubusercontent.com/85356635/129233390-909b1cbd-c29d-4472-ac67-c84ae1963323.png)

**Comment by Jinotad** [comment]
At this time, I have fired both the cook and assistant, removed all mins regarding food, hit the "reset to default" on the food tab.  Logged of the server for about 15-20 minutes, got on, and rehired the cook and assistant.  Perhaps I've corrupted my restaurant?

https://user-images.githubusercontent.com/85356635/129252008-9c2c2da3-cd74-4f09-af33-8f5b34f74b61.mp4
